crates/metrics/src/tx_status_manager_metrics.rs

```diff
@@ -0,0 +1,58 @@
+use crate::global_registry;
+use prometheus_client::metrics::gauge::Gauge;
+use std::sync::OnceLock;
+
+pub struct TxStatusManagerMetrics {
+    /// Number of prunable statuses in the manager
+    pub number_of_prunable_statuses: Gauge,
+    /// Number of non-prunable statuses in the manager
+    pub number_of_non_prunable_statuses: Gauge,
+    /// The length of the pruning queue
+    pub pruning_queue_length: Gauge,
+    /// The age of the oldest status in the pruning queue
+    pub pruning_queue_oldest_status_age_seconds: Gauge,
+}
+
+impl Default for TxStatusManagerMetrics {
+    fn default() -> Self {
+        let number_of_prunable_statuses = Gauge::default();
+        let number_of_non_prunable_statuses = Gauge::default();
+        let pruning_queue_length = Gauge::default();
+        let pruning_queue_oldest_status_age_seconds = Gauge::default();
+
+        let metrics = TxStatusManagerMetrics {
+            number_of_prunable_statuses,
+            number_of_non_prunable_statuses,
+            pruning_queue_length,
+            pruning_queue_oldest_status_age_seconds,
+        };
+
+        let mut registry = global_registry().registry.lock();
+        registry.register(
+            "tx_status_manager_number_of_prunable_statuses",
```

Comment:
the naming convention for metrics should follow -

metric_name_count if it's a number of some item
metric_name_<s/ms/ns> if it's a unit of time


```suggestion
            "tx_status_manager_prunable_status_count",
```

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2868/commits/88426455a050e117d77174f48441df3f01197a16